File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/updatetrip/StartActiveTripUseCase.java
package com.lattis.ellipse.domain.interactor.updatetrip;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.presentation.model.LockModel;
import com.lattis.ellipse.presentation.ui.ride.service.util.ServiceAction;
import com.lattis.ellipse.presentation.ui.ride.service.util.UpdateTripData;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 7/28/17.
 */

public class StartActiveTripUseCase extends UseCase<UpdateTripData>

{
    private int trip_id = 0;
    private LockModel lockModel;
    private ServiceAction serviceAction;

    @Inject
    public StartActiveTripUseCase(ThreadExecutor threadExecutor,
                                  PostExecutionThread postExecutionThread,
                                  ServiceAction serviceAction) {
        super(threadExecutor, postExecutionThread);
        this.serviceAction = serviceAction;
    }

    public StartActiveTripUseCase withTripId(int trip_id) {
        this.trip_id = trip_id;
        return this;
    }

    @Override
    protected Observable<UpdateTripData> buildUseCaseObservable() {
        return serviceAction.startActiveTripService(trip_id);
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/activetripservice/StartActiveTripUseCase.kt
package com.lattis.domain.usecase.activetripservice

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.models.UpdateTripData
import com.lattis.domain.repository.ActiveTripRepository
import com.lattis.domain.usecase.base.UseCase
import com.lattis.domain.models.Lock
import javax.inject.Inject

import io.reactivex.rxjava3.core.Observable


class StartActiveTripUseCase @Inject
constructor(threadExecutor: ThreadExecutor,
            postExecutionThread: PostExecutionThread,
            val activeTripRepository: ActiveTripRepository
) : UseCase<UpdateTripData>(threadExecutor, postExecutionThread) {
    private var trip_id = 0
    private var title:String?=null

    fun withTripId(trip_id: Int): StartActiveTripUseCase {
        this.trip_id = trip_id
        return this
    }

    fun withTitle(title: String): StartActiveTripUseCase {
        this.title = title
        return this
    }

    override fun buildUseCaseObservable(): Observable<UpdateTripData> {
        return activeTripRepository.startActiveTripService(trip_id,title!!)
    }
}
